src/test/java/uk/gov/companieshouse/web/pps/controller/pps/PenaltyRefStartsWithControllerTest.java

```diff
@@ -0,0 +1,104 @@
+package uk.gov.companieshouse.web.pps.controller.pps;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.model;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.view;
+import static uk.gov.companieshouse.web.pps.controller.pps.PenaltyRefStartsWithController.AVAILABLE_PENALTY_REF_ATTR;
+import static uk.gov.companieshouse.web.pps.controller.pps.PenaltyRefStartsWithController.PENALTY_REFERENCE_CHOICE_ATTR;
+import static uk.gov.companieshouse.web.pps.controller.pps.PenaltyRefStartsWithController.PPS_PENALTY_REF_STARTS_WITH_TEMPLATE_NAME;
+import static uk.gov.companieshouse.web.pps.util.PenaltyReference.LATE_FILING;
+import static uk.gov.companieshouse.web.pps.util.PenaltyReference.SANCTIONS;
+
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import uk.gov.companieshouse.web.pps.config.PenaltyConfigurationProperties;
+import uk.gov.companieshouse.web.pps.service.navigation.NavigatorService;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(Lifecycle.PER_CLASS)
+class PenaltyRefStartsWithControllerTest {
+
+    private static final String SELECTED_PENALTY_REFERENCE = "selectedPenaltyReference";
+    private static final String MOCK_REDIRECT = "redirect:mockControllerPath";
+
+    private MockMvc mockMvc;
+
+    @Mock
+    private NavigatorService mockNavigatorService;
+
+    private PenaltyConfigurationProperties penaltyConfigurationProperties;
+
+    @BeforeEach
+    void setup() {
+        penaltyConfigurationProperties = new PenaltyConfigurationProperties();
+        penaltyConfigurationProperties.setAllowedRefStartsWith(List.of(
+                "A", "PN"));
+        penaltyConfigurationProperties.setRefStartsWithPath(
+                "/late-filing-penalty/ref-starts-with");
+        penaltyConfigurationProperties.setEnterDetailsPath(
+                "/late-filing-penalty/enter-details");
+
+        PenaltyRefStartsWithController controller = new PenaltyRefStartsWithController(
+                mockNavigatorService,
+                penaltyConfigurationProperties);
+        mockMvc = MockMvcBuilders.standaloneSetup(controller).build();
+    }
+
+    @Test
+    void getPenaltyRefStartsWith() throws Exception {
```

Comment:
Would be good to add a displayname/description annotation to each test, make clear exactly what they're testing (or reiterate/expand on the method name).

Reply:
Done in commit [38b7726](https://github.com/companieshouse/penalty-payment-web/pull/160/commits/38b7726568166183218151ed3c0e8c16cd4f4f84)